src/main/java/com/zebrunner/agent/junit5/RerunCondition.java

```diff
@@ -64,7 +66,8 @@ private static Predicate<TestDTO> dynamicTestNeedRerun(ExtensionContext context)
     }
 
     private static Predicate<TestDTO> plainTestNeedRerun(ExtensionContext context) {
-        return test -> context.getUniqueId().equals(test.getUuid());
+        //TODO: is correlationData a valid comparison for unique id in this context?
+        return test -> context.getUniqueId().equals(test.getCorrelationData());
```

Comment:
I don't remember peculiarities of the junit agent, but at first glance it looks valid.